phac-nml/irida-next
# Clone Samples UI

## What does this PR do and why?
The front-end for cloning samples.
Fixes #404.

## Screenshots or screen recordings
The clone dialog:
![Screenshot from 2024-02-19 15-37-50](https://github.com/phac-nml/irida-next/assets/325703/36734293-91d6-47a4-bf9f-36d678f327f3)
A success message:
![notification](https://github.com/phac-nml/irida-next/assets/325703/5a2af0cc-68dc-4f9d-a758-2ff869583a5b)
A validation error:
![Screenshot from 2024-02-27 10-50-31](https://github.com/phac-nml/irida-next/assets/325703/e6bb1af3-c744-46bd-badd-5f3eb80e4702)
Sample errors:
![Screenshot from 2024-02-29 10-33-50](https://github.com/phac-nml/irida-next/assets/325703/fc6e83c6-3530-4c66-a128-0b2e5631811d)

## How to set up and validate locally
1. Navigate to a project that the logged-in user is at least a maintainer of.
1. Hit `Samples` on the left side menu.
1. Check the samples you wish to clone.
1. Click the `Clone samples` button.
1. A dialog should appear. Select the project you wish to clone the samples to.
1. Verify the samples were cloned successfully.

Note: The samples that could not be cloned should remain checked within the samples table.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
